TF-3608 Adding rule to check for and replace @action.bound

joecan opened the pull request:
##### *To:*
@jjinux @kevinzang @inancsevinc @team-f

##### *What:*
This is a new rule which checks for the use of `@action.bound` and replaces it with
`@autobind @action`.

##### *JIRA:*
https://udemyjira.atlassian.net/browse/TF-3608

##### *What did you test:*
Ran `yarn lint` in js-tooling. Also copied the rules to my node_modules in udemy-django, added some @action.bound, and ran `yarn lint-fix`.

##### *What dashboards will you be monitoring:*
N/A

joecan committed:
Adding rule to check for and replace @action.bound

kevinzang commented on packages/eslint-plugin-udemy/rules/no-action-bound/README.md:
```diff
@@ -0,0 +1,29 @@
+# No @action.bound
+
+`udemy/no-action-bound` rule checks for any uses of `@action.bound` and
+provides a fix using two seperate decorators: `@autobind @action`.
```
seperate -> separate

kevinzang reviewed (approved):
@joecan LGTM!

joecan committed:
fix: change location to decorator instead of method definition

Commit:
Publish

 - babel-polyfill-udemy-website@7.0.5
 - babel-preset-udemy-website@8.0.5
 - eslint-config-udemy-basics@5.0.1
 - eslint-config-udemy-website@7.1.3
 - eslint-plugin-udemy@6.0.1

jjinux commented on packages/babel-polyfill-udemy-website/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "babel-polyfill-udemy-website",
-  "version": "7.0.4",
```
It wasn't really a bug fix release. I was more of a new feature release. Oh well.

jjinux commented on packages/eslint-config-udemy-website/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-config-udemy-website",
-  "version": "7.1.2",
+  "version": "7.1.3",
```
I guess this should have gone to 7.2.0, but no big deal.

jjinux commented on packages/eslint-plugin-udemy/CHANGELOG.md:
```diff
@@ -3,7 +3,18 @@
+
+### Bug Fixes
+
+* change location to decorator instead of method definition ([968d25e](https://github.com/udemy/js-tooling/commit/968d25e))
```
Remember to start your Git commits with an upper-case letter, and the first line shouldn't end in a period. No biggie.

jjinux commented on packages/eslint-plugin-udemy/rules/no-action-bound/README.md:
````diff
@@ -0,0 +1,29 @@
+
+```
+
+The following patterns are not considered warnings:
````
We generally don't use warnings because everyone always ignores them. Everything is either an error or allowed.

jjinux commented on packages/eslint-plugin-udemy/rules/no-action-bound/README.md:
````diff
@@ -0,0 +1,29 @@
+The following patterns are considered warnings:
+
+```js
+
````
Do you need these blank lines around the code snippets?

jjinux commented on packages/eslint-plugin-udemy/rules/no-action-bound/tests.js:
```diff
@@ -0,0 +1,46 @@
+            }],
+        },
+    ],
+});
```
Is there any way to test the fixer?

jjinux reviewed (commented):
LGTM
